frontend/public/components/dashboards-page/overview-dashboard/inventory-card.tsx

```diff
@@ -8,128 +8,117 @@ import {
-      item.properties.additionalResources.forEach(ar => allResources.push(uniqueResource(ar, index)));
-    }
-  });
-  return allResources;
-};
+const getFirehoseResource = (model: K8sKind) => ({
+  isList: true,
+  kind: model.crd ? referenceForModel(model) : model.kind,
+  prop: 'resource',
+});
 
-const InventoryCard_: React.FC<DashboardItemProps & WithFlagsProps> = ({
+const ClusterInventoryItem = withDashboardResources(React.memo(({
   watchK8sResource,
   stopWatchK8sResource,
   resources,
-  flags = {},
-}) => {
+  model,
+  mapper,
+  useAbbr,
+  additionalResources,
+  expandedComponentLoader,
+}: ClusterInventoryItemProps) => {
   React.useEffect(() => {
-    const resourcesToWatch = getResourcesToWatch(flags);
-    resourcesToWatch.forEach(r => watchK8sResource(r));
-
+    const resource = getFirehoseResource(model);
+    watchK8sResource(resource);
+    if (additionalResources) {
+      additionalResources.forEach(watchK8sResource);
+    }
     return () => {
-      resourcesToWatch.forEach(r => stopWatchK8sResource(r));
+      stopWatchK8sResource(resource);
+      if (additionalResources) {
+        additionalResources.forEach(stopWatchK8sResource);
+      }
     };
-    // TODO: to be removed: use JSON.stringify(flags) to avoid deep comparison of flags object
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [watchK8sResource, stopWatchK8sResource, JSON.stringify(flags)]);
+  }, [watchK8sResource, stopWatchK8sResource, model, additionalResources]);
 
-  const nodesLoaded = _.get(resources.nodes, 'loaded');
-  const nodesLoadError = _.get(resources.nodes, 'loadError');
-  const nodesData = _.get(resources.nodes, 'data', []) as K8sResourceKind[];
+  const resourceData = _.get(resources.resource, 'data') as K8sResourceKind[];
+  const resourceLoaded = _.get(resources.resource, 'loaded');
+  const resourceLoadError = _.get(resources.resource, 'loadError');
 
-  const podsLoaded = _.get(resources.pods, 'loaded');
-  const podsLoadError = _.get(resources.pods, 'loadError');
-  const podsData = _.get(resources.pods, 'data', []) as PodKind[];
+  const additionalResourcesData = {};
+  let additionalResourcesLoaded = true;
+  let additionalResourcesLoadError = false;
+  if (additionalResources) {
+    additionalResourcesLoaded = additionalResources.every(r => _.get(resources[r.prop], 'loaded'));
+    additionalResources.forEach(r => {
+      additionalResourcesData[r.prop] = _.get(resources[r.prop], 'data');
+    });
+    additionalResourcesLoadError = additionalResources.some(r => !!_.get(resources[r.prop], 'loadError'));
+  }
 
-  const pvcsLoaded = _.get(resources.pvcs, 'loaded');
-  const pvcsLoadError = _.get(resources.pvcs, 'loadError');
-  const pvcsData = _.get(resources.pvcs, 'data', []) as K8sResourceKind[];
+  const ExpandedComponent = React.useCallback(() =>
+    <AsyncComponent loader={expandedComponentLoader} resource={resourceData} additionalResources={additionalResourcesData} />
```

Comment:
Nit: the line seems too long, please split the JSX expression onto multiple lines.

Reply:
fixed